src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -134,6 +134,12 @@ internal static bool VerifyAssetsAndMSBuildFilesAndPackagesArePresent(RestoreReq
                     request.Log.LogVerbose(string.Format(CultureInfo.CurrentCulture, Strings.Log_PropsFileNotOnDisk, request.Project.Name, propsFilePath));
                     return false;
                 }
+                var packageLockFilePath = PackagesLockFileUtilities.GetNuGetLockFilePath(request.Project);
```

Comment:
This is not correct. How do you know if user requested PackagesWithLockFile in the first place? 
You'd need to check whether the NuGet lock file is enabled first before doing this check. 

Furthermore you need to distinguish between the current and last restore and the no-op cache is the only thing that floats among restores, so you'd need to add something there. 

I'd be surprised if there are no test failures caused by this change.

Reply:
Thanks. Fixed.